serveradmin/serverdb/querybuilder.py

```diff
@@ -1,31 +1,314 @@
-from serveradmin.dataset.filters import Any, BaseFilter, ExactMatch
-from serveradmin.serverdb.models import Attribute
+from __future__ import unicode_literals
+
+from decimal import Decimal
+
+from adminapi.filters import (
+    AndOr,
+    Any,
+    BaseFilter,
+    Between,
+    Comparison,
+    Empty,
+    ExactMatch,
+    FilterValueError,
+    InsideNetwork,
+    InsideOnlyNetwork,
+    Overlap,
+    Regexp,
+    Startswith,
+    Not,
+)
+from serveradmin.serverdb.models import Attribute, Server, ServerAttribute
 
 
 class QueryBuilder(object):
     def __init__(self, servertypes, filters):
         self.servertypes = servertypes
-        self.sql_where = [
-            Any(*(s.pk for s in servertypes))
-            .as_sql_expr(Attribute.objects.get(pk='servertype'), servertypes)
-        ]
-        for attribute, value in filters.items():
-            if not isinstance(value, BaseFilter):
-                value = ExactMatch(value)
-            self.sql_where.append(value.as_sql_expr(attribute, servertypes))
+        self.filters = filters
 
     def build_sql(self):
-        sql = []
-        sql.append(
+        sql = (
             'SELECT'
-            '   server.server_id,'
-            '   server.hostname,'
-            '   server.intern_ip,'
-            '   server.servertype_id AS _servertype_id,'
-            '   server.project_id AS _project_id'
+            ' server.server_id,'
+            ' server.hostname,'
+            ' server.intern_ip,'
+            ' server.servertype_id AS _servertype_id,'
+            ' server.project_id AS _project_id'
+            ' FROM server'
+        )
+        sql += ' WHERE ' + self.get_sql_condition(
+            Attribute.objects.get(pk='servertype'),
+            Any(*(s.pk for s in self.servertypes)),
+        )
+        for attribute, value in self.filters.items():
+            sql += ' AND ' + self.get_sql_condition(attribute, value)
+
+        return sql
+
+    def get_sql_condition(self, attribute, value):  # NOQA C901
```

Comment:
That's too much new stuff for to review in whole now. Would have suggest to do all this new stuff in a separated commit.

Reply:
Fair enough.  It is kinda separate.  I will simplify this function later.